mleap-xgboost-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/mleap/XGBoostClassificationModelParitySpec.scala

```diff
@@ -136,32 +159,39 @@ class XGBoostClassificationModelParitySpec extends FunSpec
       for(bf <- managed(BundleFile(file))) {
         transformer.writeBundle.format(SerializationFormat.Json).save(bf).get
       }
-
-      bundleCache = Some(file)
       file
     }
   }
 
-  def mleapTransformer(transformer: Transformer)
+  def mleapTransformer(transformer: Transformer, dataset: DataFrame, bundleCache: Option[File])
                       (implicit context: SparkBundleContext): frame.Transformer = {
     import ml.combust.mleap.runtime.MleapSupport._
-
-    (for(bf <- managed(BundleFile(serializedModel(transformer)))) yield {
+    val file = serializedModel(transformer, dataset, bundleCache)
+    (for(bf <- managed(BundleFile(file))) yield {
       bf.loadMleapBundle().get.root
     }).tried.get
   }
 
-  it("produces the same results") {
+  def doTest(sparkTransformer: Transformer, dataset: DataFrame, bundleCache: Option[File]): Unit ={
     val sparkDataset = sparkTransformer.transform(dataset)
     val mleapSchema = TypeConverters.sparkSchemaToMleapSchema(dataset)
 
     val data = dataset.collect().map {
       r => Row(r.toSeq: _*)
     }
     val frame = DefaultLeapFrame(mleapSchema, data)
-    val mleapT = mleapTransformer(sparkTransformer)
+    val mleapT = mleapTransformer(sparkTransformer, dataset, bundleCache)
     val mleapDataset = mleapT.transform(frame).get
```

Comment:
this should ideally be one responsibility

Reply:
> @talalryz Do you know why tests are not passing ?

the tests are not passing because xgboost 0.90 does not allow Float.NaN as missing value for  sparse Vectors.  

> ERROR DataBatch: java.lang.RuntimeException: you can only specify missing value as 0.0 (the currently set value NaN) when you have SparseVector or Empty vector as your feature format

This is expected as this branch does not fix the issues introduced by the recent upgrade to xgboost 0.90.

Reply:
updated the branch so that all tests pass